controller/setting_controller.go

```diff
@@ -403,6 +428,90 @@ func (sc *SettingController) updatePriorityClass() error {
 	return nil
 }
 
+func (sc *SettingController) updateDefaultStorageClass() error {
+	storageConfigMap, err := sc.ds.GetConfigMapRO(types.DefaultStorageClassConfigMapName)
+	if err != nil {
+		if datastore.ErrorIsNotFound(err) {
+			// The storageclass configmap is deleted therefore we delete the actual storageclass
+			err := sc.ds.DeleteStorageClass(types.DefaultStorageClassName)
+			if err != nil && !datastore.ErrorIsNotFound(err) {
+				return nil
+			}
+			return nil
+		}
+		return err
+	}
+
+	storageclass, err := buildStorageClassManifestFromConfigMap(storageConfigMap)
+	if err != nil {
+		return err
```

Comment:
So a failure to parse the config map will stop the setting controller from reconciliation?

Also, the extra setting name is weird. Now I think it's better to have `ConfigMapController` to take in the change instead.

Reply:
Yes, I agree that the extra setting is a bit weird.